### PR TITLE
63 Use $? in shell script for compatibility

### LINKED ISSUE #63: tmux-complete 0.1.0 not working with neocomplete
I've updated from tmux-complete 0.0.7 on 0.1.0 and the completion is not working anymore. Here is my minimal vimrc:

``` vim
set nocompatible " Be iMproved

let g:neosnippet#expand_word_boundary = 1
let g:plug_threads = 40

call plug#begin('~/.vim/plugged/') 

Plug 'Shougo/neocomplete.vim'
Plug 'wellle/tmux-complete.vim', 'v0.1.0'

call plug#end()

let g:neocomplete#enable_at_startup = 1
" Use smartcase.
let g:neocomplete#enable_smart_case = 1
" Set minimum syntax keyword length.
let g:neocomplete#sources#syntax#min_keyword_length = 3
" Fixing an autocomplete bug with vim-notes
let g:neocomplete#force_overwrite_completefunc=1

" Define keyword.
if !exists('g:neocomplete#keyword_patterns')
	let g:neocomplete#keyword_patterns = {}
endif
let g:neocomplete#keyword_patterns['default'] = '\h\w*'

" Plugin key-mappings.
inoremap <expr><C-g> neocomplete#undo_completion()

" Enable omni completion.
autocmd FileType css setlocal omnifunc=csscomplete#CompleteCSS
autocmd FileType html setlocal omnifunc=htmlcomplete#CompleteTags
autocmd FileType javascript setlocal omnifunc=javascriptcomplete#CompleteJS
autocmd FileType ruby setlocal omnifunc=rubycomplete#Complete
autocmd FileType xml setlocal omnifunc=xmlcomplete#CompleteTags

" Enable heavy omni completion.
if !exists('g:neocomplete#sources#omni#input_patterns')
	let g:neocomplete#sources#omni#input_patterns = {}
endif

if !exists('g:neocomplete#force_omni_input_patterns')
	let g:neocomplete#force_omni_input_patterns = {}
endif

" Prevent from pressing <C-x><C-o> for file aware autocompletion in ruby
let g:neocomplete#force_omni_input_patterns.ruby = '[^.  *\t]\.\w*\|\h\w*::' 

let g:tmuxcomplete#trigger = ''
```

I tried different things in the `g:tmuxcomplete#trigger` but that didn't work. If I switch back to tmux-complete 0.0.7 everything works. I have tmux 2.1 on my xubuntu 14.10 machine.

### PR DESCRIPTION
**Pull request by wellle**
Close #63

### CODE REVIEW

File: sh/tmuxcomplete [hunk 1]
```diff
@@ -27,7 +27,8 @@
 # Words including 2000 lines of history per pane
 #     sh tmuxcomplete -c '-S -2000'
 
-if ! tmux info &> /dev/null; then
```
Comment (justinmk) [inline sh/tmuxcomplete]: I thought this was a POSIX pattern. What version of which shell is not compatible with it? (It's not clear to me after reading https://github.com/wellle/tmux-complete.vim/issues/63 )

Comment (wellle) [inline sh/tmuxcomplete]: To be honest I'm not quite sure. I'm not able to reproduce the issue, but this change seemed to work for @wikimatze. Feel free to join the discussion in #63. I'd like to get to the bottom of this as well.

Comment (justinmk) [inline sh/tmuxcomplete]: @wellle  Not sure if it's the cause of the issue, but I am pretty sure `&>` is a bashism. The POSIX form would be `2>&1 >/dev/null`.

File: sh/tmuxcomplete [hunk 2]
```diff
@@ -27,7 +27,8 @@
...
+tmux info &> /dev/null
+if [ "$?" != "0" ]; then
```
Comment (justinmk) [inline sh/tmuxcomplete]: `!=` also is not POSIX, it requires bash double-brackets `[[ ]]`. POSIX would be:

    if ! [ "$?" = "0" ]; then

or I would tend to use:

    if [ $? -ne 0 ] ; then

File: sh/tmuxcomplete (hunk 1 above)
Comment (wellle) [inline sh/tmuxcomplete]: Oh you might be onto something. @wikimatze: Could you give that a try on clean v0.1.0?

File: sh/tmuxcomplete (hunk 2 above)
Comment (wellle) [inline sh/tmuxcomplete]: Thanks a lot for helping me avoid follow up mistakes. Is there some kind of tool to check for POSIX compatibility?

I will update this tomorrow :+1:

Comment (justinmk) [inline sh/tmuxcomplete]: https://github.com/koalaman/shellcheck is awesome though I don't actually use it much.

**Comment by wellle** [comment]
@justinmk: Thanks for your help!

I installed dash locally to test for POSIX compatibility. I managed to reproduce the issue on master and tried your suggestions. The bashism `&>` was indeed the issue, and using `2>&1` instead solved it. I had to put it after `> /dev/null` though to properly redirect stderr (see updated diff).

I also found that `!=` is in fact POSIX compliant, see:
http://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html.

**Pull request merged**